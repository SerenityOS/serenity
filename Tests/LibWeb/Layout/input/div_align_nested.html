<style>
    .square {
        width: 100px; 
        height: 100px;
        background-color:green
    }
</style>


<div align="right">This text and the green square are both right aligned: <div class="square"></div>
    <div align="left">
        This text and the green square are both left aligned despite being nested in a div with align="right":
        <div class="square"/>
    </div>
</div>
