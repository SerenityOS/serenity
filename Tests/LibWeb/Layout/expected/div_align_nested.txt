Viewport <#document> at (0,0) content-size 800x600 children: not-inline
  BlockContainer <html> at (0,0) content-size 800x600 [BFC] children: not-inline
    BlockContainer <body> at (8,8) content-size 784x252.875 children: not-inline
      BlockContainer <div> at (8,8) content-size 784x252.875 children: not-inline
        BlockContainer <(anonymous)> at (8,8) content-size 784x17.46875 children: inline
          line 0 width: 447.484375, height: 17.46875, bottom: 17.46875, baseline: 13.53125
            frag 0 from TextNode start: 0, length: 54, rect: [345,8 447.484375x17.46875]
              "This text and the green square are both right aligned:"
          TextNode <#text>
        BlockContainer <div.square> at (692,25.46875) content-size 100x100 children: not-inline
        BlockContainer <(anonymous)> at (8,125.46875) content-size 784x0 children: inline
          TextNode <#text>
        BlockContainer <div> at (8,125.46875) content-size 784x135.40625 children: not-inline
          BlockContainer <(anonymous)> at (8,125.46875) content-size 784x35.40625 children: inline
            line 0 width: 711.4375, height: 17.46875, bottom: 17.46875, baseline: 13.53125
              frag 0 from TextNode start: 1, length: 87, rect: [8,125.46875 711.4375x17.46875]
                "This text and the green square are both left aligned despite being nested in a div with"
            line 1 width: 94.296875, height: 17.9375, bottom: 35.40625, baseline: 13.53125
              frag 0 from TextNode start: 89, length: 14, rect: [8,142.46875 94.296875x17.46875]
                "align="right":"
            TextNode <#text>
          BlockContainer <div.square> at (8,160.875) content-size 100x100 children: inline
            TextNode <#text>
          BlockContainer <(anonymous)> at (8,260.875) content-size 784x0 children: inline
            TextNode <#text>
        BlockContainer <(anonymous)> at (8,260.875) content-size 784x0 children: inline
          TextNode <#text>

ViewportPaintable (Viewport<#document>) [0,0 800x600]
  PaintableWithLines (BlockContainer<HTML>) [0,0 800x600]
    PaintableWithLines (BlockContainer<BODY>) [8,8 784x252.875] overflow: [8,8 784.484375x252.875]
      PaintableWithLines (BlockContainer<DIV>) [8,8 784x252.875] overflow: [8,8 784.484375x252.875]
        PaintableWithLines (BlockContainer(anonymous)) [8,8 784x17.46875] overflow: [8,8 784.484375x17.46875]
          TextPaintable (TextNode<#text>)
        PaintableWithLines (BlockContainer<DIV>.square) [692,25.46875 100x100]
        PaintableWithLines (BlockContainer(anonymous)) [8,125.46875 784x0]
        PaintableWithLines (BlockContainer<DIV>) [8,125.46875 784x135.40625]
          PaintableWithLines (BlockContainer(anonymous)) [8,125.46875 784x35.40625]
            TextPaintable (TextNode<#text>)
          PaintableWithLines (BlockContainer<DIV>.square) [8,160.875 100x100]
          PaintableWithLines (BlockContainer(anonymous)) [8,260.875 784x0]
        PaintableWithLines (BlockContainer(anonymous)) [8,260.875 784x0]
